openedx/adg/lms/student/views.py

```diff
@@ -64,3 +73,120 @@ def get_social_oauth_providers_data(self):
+
+        completed_courses_details, inprogress_courses_details = self.get_courses_overview_and_status()
+
+        webinars = self.get_webinars_overview()
+
+        certificates = self.get_certificates_overview()
+
+        context.update({
+            'user': self.request.user,
+            'completed_courses': completed_courses_details,
+            'inprogress_courses': inprogress_courses_details,
+            'webinars': webinars,
+            'certificates': certificates,
+        })
+        return context
+
+    def get_courses_overview_and_status(self):
+        """
+            Returns: list of completed and in progress course(s) with details
+        """
+        enrolled_course_ids = list(
+            CourseEnrollment.objects.filter(user=self.request.user).values_list('course', flat=True)
+        )
+        completed_courses = list(
+            PersistentCourseGrade.objects.filter(Q(user_id=self.request.user.id) and ~Q(passed_timestamp=None)).values(
+                'course_id', 'percent_grade')
+        )
+
+        completed_course_ids = [completed_courses[index]['course_id'] for index in range(len(completed_courses))]
+        completed_course_grades = [completed_courses[index]['percent_grade'] for index in
+                                   range(len(completed_courses))]
+
+        courses_overview = list(
+            CourseOverview.objects.filter(
+                id__in=enrolled_course_ids
+            ).values('id', 'display_name', 'short_description', 'course_image_url')
+        )
+
+        completed_courses_details = []
+        inprogress_courses_details = []
+
+        for course_overview in courses_overview:
+            course_overview.update({
+                'details_url': reverse(
+                    course_home_url_name(course_overview['id']),
+                    args=[six.text_type(course_overview['id'])]
+                )
+            })
+
+            if course_overview['id'] in completed_course_ids:
+                course_overview.update({
+                    'status': 'Completed',
+                    'grade': completed_course_grades[completed_course_ids.index(course_overview['id'])]
+                })
+                completed_courses_details.append(course_overview)
+            else:
+                course_overview.update({
+                    'status': 'In Progress',
+                    'grade': None
+                })
+                inprogress_courses_details.append(course_overview)
+
+        return completed_courses_details, inprogress_courses_details
+
+    def get_webinars_overview(self):
+        """
+            Returns: list of user's registered webinar(s) with details
+        """
+        webinar_ids = list(
+            WebinarRegistration.objects.filter(email=self.request.user.email).values_list('webinar', flat=True))
+
+        webinars = list(
+            Webinar.objects.filter(id__in=webinar_ids).values(
```

Comment:
You can directly get the webinars from WebinarRegistration model because it has relation with Webinar

Reply:
There is no need to filter out the Webinar again.